src/Utils.php

```diff
@@ -381,4 +381,18 @@ private static function idnToAsci(string $domain, int $options, ?array &$info =
 
         throw new \Error('ext-idn or symfony/polyfill-intl-idn not loaded or too old');
     }
+
+    /**
+     * Redacts the password in the userinfo part of a UriInterface.
+     */
+    public static function redactUserinfoInUri(UriInterface $uri): UriInterface
```

Comment:
Probably this function should go in the PSR package. We can add it to only 2.x, and drop support for psr 1.x in guzzle 7.9.

Reply:
Alternatively, we could view this as a breaking change, and just do it in Guzzle 8.

Reply:
I consider this change a bug fix rather than a new feature (see also the PR description that this technically is a security fix). So it should probably go into Guzzle 7.8. 

Nevertheless I created a PR guzzle/psr7#607

Should I mark this function in guzzle/guzzle as `@internal` until it can be replaced by the one in PSR-7?